dvc/remote/base.py

```diff
@@ -588,13 +588,13 @@ def download(
     def _download_dir(
         self, from_info, to_info, name, no_progress_bar, file_mode, dir_mode
     ):
-        file_to_infos = (
-            to_info / file_to_info.relative_to(from_info)
-            for file_to_info in self.walk_files(from_info)
+        from_infos = list(self.walk_files(from_info))
```

Comment:
There might a significant delay here before we see pbar. May ignore this for now though.

Reply:
@Suor Very good point! It has similar reasons to what @shcheklein brought up in his `-d s3://path/to/dir` research(which is _collect_dir's fault). I'll fix it in https://github.com/iterative/dvc/issues/2911 .